### ISSUE TITLE
Compatibility Issue: Crash on Server Startup with Mixin Error when mods such as carpet-tis addition or Easyauth

### ISSUE THREAD

**Issue by tryptophawa** [senseiwells/ServerReplay#62]
**Issue:** Crash on Server Startup with Mixin Error after Prime Backup (PB) Restore
**Description:**
After restoring a backup using `Prime Backup`, the server crashes during startup due to a mixin error. This issue might cause by conflict between the order of PB restore and chunk recording
**Error Message:**
```
Critical problem: server_translations_api.mixin.json:packet.TranslatableTextContentMixin from mod server_translations_api target net.minecraft.class_2588 was loaded too early.
```
**Steps to Reproduce:**
1. Take a backup of the server using Prime Backup.
2. Open chunk recording in `config.json`
3. Restore the backup using Prime Backup.
4. Start the server.
5. Observe the crash with the above error message.
**Expected Behavior:**
The server should start without any issues after restoring the backup, and the recorded actions should be record correctly.
**Environment:**
- Minecraft Version: 1.20.4
- Fabric Loader: 0.16.2
- Mixin Subsystem: 0.8.7
- Mods: Carpet, Lithium, Memoryleakfix, Server-Replay, and more.
- MCDR: MCDreforged 2.13.1
- PB: PB v1.8.1
**Additional Context:**
The issue arises because the restored world state does not match the state that `server-replay` requires. When the server starts, `server-replay` attempts to replay the recorded actions but world isn't ready for recording.
**Temperary fix:**
Disable the automatic recording of blocks in the `server-replay` mod before or after restoring the backup. This can typically be done through the mod's configuration file or by using a command in-game.
**Possible way to fix:**
start recording chunk later when world already generated.

I will attach logs and configs later.

**Comment by senseiwells** [comment]
From the information you have provided I do not think that ServerReplay is responsible for your crash.

Your error is referring to `server_translations_api`, which is not used in ServerReplay. But without further logs this information is not very useful.

ServerReplay doesn't care for what the world state is, it just records whatever is present.

**Comment by tryptophawa** [comment]
[latest-crash.log](https://github.com/user-attachments/files/17024272/latest.log)
[config-crash.json](https://github.com/user-attachments/files/17024273/config.json)
log of crash (duplicate just now) and config for replay
**when I remove the chunk replay the server can start succesfully**
[latest-success.log](https://github.com/user-attachments/files/17024331/latest.1.log)
[config-success.json](https://github.com/user-attachments/files/17024323/config.1.json)

**Comment by tryptophawa** [comment]
[config.zip](https://github.com/user-attachments/files/17024394/config.zip)
[mods-1.zip](https://github.com/user-attachments/files/17024466/mods-1.zip)
[mods-2.zip](https://github.com/user-attachments/files/17024470/mods-2.zip)
all mods and configs are here if you needed

**Comment by senseiwells** [comment]
This doesn't seem like it is caused by ServerReplay. The crash is happening at the 'preLaunch' stage which is before ServerReplay has even loaded the config file.

```
[15:12:04] [main/ERROR]: A mod crashed on startup!
net.fabricmc.loader.impl.FormattedException: java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'carpet-tis-addition' at 'carpettisaddition.CarpetTISAdditionPreLaunch'!
```

Could you try reproducing the crash with only ServerReplay and fabric-api installed?

It is likely a different mod causing this crash, either carpet-tis or EasyAuth. Also try removing those and see if the crash still happens.

**Comment by tryptophawa** [comment]
If the server only has server replay and fabric-API, there is no problem with starting up the server.
But if I add those mods above, I have to turn off chunk replay.
Maybe it is a compatibility issue with mods such as carpet-tis or EasyAuth? Since the server starts correctly without chunk replay.

**Comment by senseiwells** [comment]
Yes, it's likely one of those mods, I would keep adding mods to your mods list until your server crashes, then the last mod you added is likely causing the issue. Let me know which mod it is, and then you should probably also report this issue to them.

**Comment by HoldYourWaffle** [comment]
I'm getting the same error when combining [Armor Stand Editor](https://modrinth.com/mod/armorstandeditor) with Server Replay, also only if chunk recordings are configured: [crash.log](https://github.com/user-attachments/files/17129452/crash.log)

I've only touched mixins once or twice and never seen an error like this, but given the nature of the error message, could it be that Server Replay's chunk recording mechanism somehow loads some classes when they aren't supposed to yet, e.g. during the static initialization of a class?

**Comment by senseiwells** [comment]
Cannot reproduce: https://mclo.gs/WTtPeZa

Please provide your exact config

**Comment by senseiwells** [comment]
Please also update to the latest version of the mod

**Comment by senseiwells** [comment]
I believe this was fixed by chance in 1.1.3 or 1.1.4, update your version of ServerReplay, if the issue persists please re-open this issue with a copy of your config.

**Comment by HoldYourWaffle** [comment]
Yep, can confirm it works with v1.1.4. Thanks a lot!
